src/GafferAppleseed/IECoreAppleseedPreview/Renderer.cpp

```diff
@@ -1693,6 +1713,201 @@ class AppleseedDeltaLight : public AppleseedLight
 		asf::Transformd m_transform;
 };
 
+/// Appleseed area light handle.
+class AppleseedAreaLight : public AppleseedLight
+{
+	public :
+
+		AppleseedAreaLight( asr::Project &project, const string &name, const IECoreScenePreview::Renderer::AttributesInterface *attributes, IECoreScenePreview::Renderer::RenderType renderType )
+			:	AppleseedLight( project, name, attributes, renderType == IECoreScenePreview::Renderer::Interactive ) , m_transform( asf::Transformd::identity() ), m_renderType( renderType )
+		{
+			init();
+			AppleseedAreaLight::attributes( attributes );
+		}
+
+		virtual ~AppleseedAreaLight()
+		{
+			if( isInteractiveRender() )
+			{
+				removeAreaLightEntities();
+			}
+			else
+			{
+				// Create the material assignments.
```

Comment:
> I've been considering changing things so that the object() call is passed everything upfront (the transform as well as the attributes) so that batch renderers can do everything in one shot. We'd all return a null pointer unless the render mode was interactive, so there's no temptation to try to make edits during a batch generation. Do you think that would make sense?

Yes, I think it would make sense. I could simplify some parts of the code if I had all the information in object().